tests/test_extract_model_langchain_openai.py

```diff
@@ -47,3 +49,254 @@ def test_entire_llm_call_using_langchain_openai(expected_model, model):
+@pytest.mark.parametrize(  # noqa: F821
+    "expected_model,model",
+    [
+        ("gpt-3.5-turbo", ChatOpenAI()),
+        ("gpt-3.5-turbo-instruct", OpenAI()),
+        (
+            "gpt-3.5-turbo",
+            AzureChatOpenAI(
+                openai_api_version="2023-05-15",
+                azure_deployment="your-deployment-name",
+                azure_endpoint="https://your-endpoint-name.azurewebsites.net",
+            ),
+        ),
+        (
+            "gpt-3.5-turbo-instruct",
+            AzureOpenAI(
+                openai_api_version="2023-05-15",
+                azure_deployment="your-deployment-name",
+                azure_endpoint="https://your-endpoint-name.azurewebsites.net",
+            ),
+        ),
+    ],
+)
+async def test_ainvoke_on_llm_call_using_langchain_openai(expected_model, model):
+    callback = CallbackHandler()
+    try:
+        # LLM calls are failing, because of missing API keys etc.
+        # However, we are still able to extract the model names beforehand.
+        await model.ainvoke("Hello, how are you?", config={"callbacks": [callback]})
+    except Exception as e:
+        print(e)
+        pass
+
+    callback.flush()
+    api = get_api()
+
+    trace = api.trace.get(callback.get_trace_id())
+    assert len(trace.observations) == 1
+
+    generation = list(filter(lambda o: o.type == "GENERATION", trace.observations))[0]
+    assert generation.model == expected_model
+
+
+@pytest.mark.parametrize(  # noqa: F821
+    "expected_model,model",
+    [
+        ("gpt-3.5-turbo", ChatOpenAI(streaming=True)),
+        ("gpt-3.5-turbo-instruct", OpenAI(streaming=True)),
+        (
+            "gpt-3.5-turbo",
+            AzureChatOpenAI(
+                openai_api_version="2023-05-15",
+                azure_deployment="your-deployment-name",
+                azure_endpoint="https://your-endpoint-name.azurewebsites.net",
+                streaming=True,
+            ),
+        ),
+        (
+            "gpt-3.5-turbo-instruct",
+            AzureOpenAI(
+                openai_api_version="2023-05-15",
+                azure_deployment="your-deployment-name",
+                azure_endpoint="https://your-endpoint-name.azurewebsites.net",
+                streaming=True,
+            ),
+        ),
+    ],
+)
+def test_simple_streaming_llm_call_with_langchain_openai(expected_model, model):
+    callback = CallbackHandler()
+    try:
+        res = model.stream("Hello, how are you?", config={"callbacks": [callback]})
+        for chunk in res:
+            chunk
+    except Exception as e:
+        print(e)
+        pass
+
+    callback.flush()
+    api = get_api()
+
+    trace = api.trace.get(callback.get_trace_id())
+
+    assert _is_streaming_response(res)
+    assert len(trace.observations) == 1
+
+    generation = list(filter(lambda o: o.type == "GENERATION", trace.observations))[0]
+    assert generation.model == expected_model
+
+
+def _is_streaming_response(response):
```

Comment:
Minor: you can import this function also from https://github.com/noble-varghese/langfuse-python/blob/318f3238c0dcb7c0bf361c0bad58433386dc46e4/langfuse/openai.py#L451